### DISCUSSION TITLE
how to desine a workflow for this scene.

### DISCUSSION THREAD

**Discussion by tianweimol** [elsa-workflows/elsa-core#3269] [Q&A]
thanks for your answer
i have a scene.
1. create a form what is descripted a model in database
2. save form data and start a workflow
3. send the form to leader ,wait for approve. the approve information contains 1.approve result(e.g approved,reject); 2. comment 3.file attachement
4. send to next leader to approve
5. if all activity is approved,the workflow will be completed

how i should desine the workflow?
thank you

**Comment by sfmskywalker** [comment]
You can implement this in different ways, but the easiest way would be to use the signaling activities to block the workflow while waiting for approval/rejection from the humans that receive e.g. emails with user tasks.

This guide can help you get started: https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-document-approval

**Reply by tianweimol** [reply]
thank you very much!
